File: src/core/markdown_fmt.py
"""Tier-1 markdown / language helpers shared across derivative publish
sidecars.

Pure functions, no derivative-type knowledge. The per-derivative publish
templates live in each derivative's tool module (tools/<x>/publish.py)
and import these helpers — keeping product policy out of core.
"""

from __future__ import annotations


def is_zh(lang_iso: str) -> bool:
    """True if `lang_iso` is a Chinese language tag (zh, zh-CN, zh-TW...)."""
    return (lang_iso or "").lower().split("-")[0].startswith("zh")


def t(lang_iso: str, zh: str, en: str) -> str:
    """Pick zh or en string based on `lang_iso`. Used by publish renderers
    to localize headings / labels to the *source video's* language, not
    the UI language."""
    return zh if is_zh(lang_iso) else en


def fmt_dur(seconds: float) -> str:
    """Seconds → `H:MM:SS` (when ≥1h) or `M:SS`."""
    s = max(0, int(seconds))
    h, rem = divmod(s, 3600)
    m, s = divmod(rem, 60)
    return f"{h:d}:{m:02d}:{s:02d}" if h else f"{m:d}:{s:02d}"


def fmt_hashtags(tags) -> str:
    """List of tag strings → single space-joined `#a #b #c` line.
    Each tag is prefixed with `#` if not already. Non-list / empty → ""."""
    if not isinstance(tags, (list, tuple)):
        return ""
    parts = []
    for tag in tags:
        s = str(tag).strip().lstrip("#")
        if s:
            parts.append("#" + s)
    return " ".join(parts)

File: desktop/src/creations/shared/markdownFmt.ts
/**
 * Tier-1 markdown / language helpers shared across derivative publish docs
 * (ADR-0008 TS port of `src/core/markdown_fmt.py`).
 *
 * Pure functions, no derivative-type knowledge. Per-derivative publish templates
 * (clip/publish.ts, news_desk/publish.ts) import these. Headings/labels localize
 * to the *source video's* language (lang_iso), not the UI language.
 */

/** True if `langIso` is a Chinese tag (zh, zh-CN, zh-TW …). */
export function isZh(langIso: string): boolean {
  return (langIso || "").toLowerCase().split("-")[0]!.startsWith("zh");
}

/** Pick zh or en based on `langIso`. */
export function t(langIso: string, zh: string, en: string): string {
  return isZh(langIso) ? zh : en;
}

/** Seconds → `H:MM:SS` (when ≥1h) or `M:SS`. */
export function fmtDur(seconds: number): string {
  const total = Math.max(0, Math.trunc(seconds));
  const h = Math.trunc(total / 3600);
  const m = Math.trunc((total % 3600) / 60);
  const s = total % 60;
  const pad = (n: number) => String(n).padStart(2, "0");
  return h ? `${h}:${pad(m)}:${pad(s)}` : `${m}:${pad(s)}`;
}

/** List of tag strings → single space-joined `#a #b #c`. Each tag gets a `#`
 *  if not already present. Non-array / empty → "". */
export function fmtHashtags(tags: unknown): string {
  if (!Array.isArray(tags)) return "";
  const parts: string[] = [];
  for (const tag of tags) {
    const s = String(tag).trim().replace(/^#+/, "");
    if (s) parts.push("#" + s);
  }
  return parts.join(" ");
}
